### ISSUE TITLE
Issues with ec2_instance.py  The error was: TypeError: 'NoneType' object is not subscriptable

### ISSUE THREAD

**Issue by autoctrl1** [ansible-collections/amazon.aws#945]
### Summary

When trying to start or run tasks to stop or start an ec2 instance getting the following error when using the amazon.aws 3.0.0 collection

/ansible_amazon.aws.ec2_instance_payload.zip/ansible_collections/amazon/aws/plugins/modules/ec2_instance.py\\", line 1501, in get_default_subnet\\nTypeError: 'NoneType' object is not subscriptable\\n", "module_stdout": "", "msg": "MODULE FAILURE\\nSee stdout/stderr for the exact error", "rc": 1}

### Issue Type

Bug Report

### Component Name

ec2_instance

### Ansible Version

```console (paste below)
$ ansible --version

```
ansible 2.9.27

### Collection Versions

```console (paste below)
$ ansible-galaxy collection list
```
---
collections:
  - name: community.aws
    version: 2.1.0
  - name: amazon.aws
    version: 3.0.0
  - name: community.general
    version: 4.2.0
  - name: ansible.windows
    version: 1.9.0
  - name: community.windows
    version: 1.9.0

### AWS SDK versions

```console (paste below)
$ pip show boto boto3 botocore
```
(python3) [/var/lib/awx/venv/python3/bin]# pip show boto boto3 botocore
Name: boto
Version: 2.49.0
Summary: Amazon Web Services Library
Home-page: https://github.com/boto/boto/
Author: Mitch Garnaat
Author-email: mitch@garnaat.com
License: MIT
Location: /var/lib/awx/venv/python3/lib/python3.6/site-packages
Requires: 
Required-by: 
---
Name: boto3
Version: 1.21.4
Summary: The AWS SDK for Python
Home-page: https://github.com/boto/boto3
Author: Amazon Web Services
Author-email: 
License: Apache License 2.0
Location: /var/lib/awx/venv/python3/lib/python3.6/site-packages
Requires: botocore, jmespath, s3transfer
Required-by: 
---
Name: botocore
Version: 1.24.5
Summary: Low-level, data-driven core of boto 3.
Home-page: https://github.com/boto/botocore
Author: Amazon Web Services
Author-email: 
License: Apache License 2.0
Location: /var/lib/awx/venv/python3/lib/python3.6/site-packages
Requires: jmespath, python-dateutil, urllib3
Required-by: awscli, boto3, s3transfer

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```
COLLECTIONS_PATHS(/etc/ansible/ansible.cfg) = [u'/opt/collections']
DEFAULT_ROLES_PATH(/etc/ansible/ansible.cfg) = [u'/etc/ansible/roles', u'/usr/share/ansible/roles']
DEFAULT_TIMEOUT(/etc/ansible/ansible.cfg) = 6

### OS / Environment

RHEL 7.9

### Steps to Reproduce

```yaml (paste below)

```
- name: Gather EC2 Facts
  ec2_instance_info:
    region: "{{ region }}"
    filters:
      private-ip-address: "{{ instance_ip }}"
      "tag:Project": "{{ project_tag | lower }}"
    aws_access_key: "{{ assumed_role.sts_creds.access_key | default(omit) }}"
    aws_secret_key: "{{ assumed_role.sts_creds.secret_key | default(omit) }}"
    security_token: "{{ assumed_role.sts_creds.session_token | default(omit) }}"
  register: instance_facts

- name: Get platform details
  set_fact:
    instance_os: "{{ instance_facts.instances[0].tags.operating_system | default('') }}"

- name: changing state of ec2 instance
  amazon.aws.ec2_instance:
    state: "{{ instance_state }}"
    instance_ids: "{{ instance_facts.instances[0].instance_id }}"
    region: "{{ region }}"
    aws_access_key: "{{ assumed_role.sts_creds.access_key | default(omit) }}"
    aws_secret_key: "{{ assumed_role.sts_creds.secret_key | default(omit) }}"
    security_token: "{{ assumed_role.sts_creds.session_token | default(omit) }}"
    wait: yes
    wait_timeout: 10000

### Expected Results

Playbook to complete the start up or stop of an instance without failure

### Actual Results

```console (paste below)

```
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\\n  File \\"/var/lib/awx/.ansible/tmp/ansible-tmp-1659107137.3688648-266-108601800665519/AnsiballZ_ec2_instance.py\\", line 102, in <module>\\n    _ansiballz_main()\\n  File \\"/var/lib/awx/.ansible/tmp/ansible-tmp-1659107137.3688648-266-108601800665519/AnsiballZ_ec2_instance.py\\", line 94, in _ansiballz_main\\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\\n  File \\"/var/lib/awx/.ansible/tmp/ansible-tmp-1659107137.3688648-266-108601800665519/AnsiballZ_ec2_instance.py\\", line 40, in invoke_module\\n    runpy.run_module(mod_name='ansible_collections.amazon.aws.plugins.modules.ec2_instance', init_globals=None, run_name='__main__', alter_sys=True)\\n  File \\"/usr/lib64/python3.6/runpy.py\\", line 205, in run_module\\n    return _run_module_code(code, init_globals, run_name, mod_spec)\\n  File \\"/usr/lib64/python3.6/runpy.py\\", line 96, in _run_module_code\\n    mod_name, mod_spec, pkg_name, script_name)\\n  File \\"/usr/lib64/python3.6/runpy.py\\", line 85, in _run_code\\n    exec(code, run_globals)\\n  File \\"/tmp/ansible_amazon.aws.ec2_instance_payload_63moo8ps/ansible_amazon.aws.ec2_instance_payload.zip/ansible_collections/amazon/aws/plugins/modules/ec2_instance.py\\", line 2015, in <module>\\n  File \\"/tmp/ansible_amazon.aws.ec2_instance_payload_63moo8ps/ansible_amazon.aws.ec2_instance_payload.zip/ansible_collections/amazon/aws/plugins/modules/ec2_instance.py\\", line 1989, in main\\n  File \\"/tmp/ansible_amazon.aws.ec2_instance_payload_63moo8ps/ansible_amazon.aws.ec2_instance_payload.zip/ansible_collections/amazon/aws/plugins/modules/ec2_instance.py\\", line 1894, in build_filters\\n  File \\"/tmp/ansible_amazon.aws.ec2_instance_payload_63moo8ps/ansible_amazon.aws.ec2_instance_payload.zip/ansible_collections/amazon/aws/plugins/modules/ec2_instance.py\\", line 1501, in get_default_subnet\\nTypeError: 'NoneType' object is not subscriptable\\n", "module_stdout": "", "msg": "MODULE FAILURE\\nSee stdout/stderr for the exact error", "rc": 1}

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibullbot** [comment]
Files identified in the description:
* [`plugins/modules/ec2_instance.py`](https://github.com/['ansible-collections/amazon.aws', 'ansible-collections/community.aws', 'ansible-collections/community.vmware']/blob/main/plugins/modules/ec2_instance.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibullbot** [comment]
cc @jillr @ryansb @s-hertel @tremble
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by tremble** [comment]
Thanks for taking the time to open this issue.  This appears to be a duplicate of #467
